test/project_types/extension/extension_test_helpers/test_extension_setup.rb

```diff
@@ -6,14 +6,21 @@ module TestExtensionSetup
       def setup
         ShopifyCli::ProjectType.load_type(:extension)
 
-        @test_extension_type = ExtensionTestHelpers::TestExtension.new
-        Models::Type.repository[@test_extension_type.identifier] = @test_extension_type
+        @_original_extension_specifications = Extension.specifications
+
+        Extension.specifications = Models::Specifications.new(
+          custom_handler_root: File.expand_path('../', __FILE__),
+          custom_handler_namespace: ::Extension::ExtensionTestHelpers,
+          fetch_specifications: -> { [Models::Specification.new(identifier: 'test_extension')] }
+        )
```

Comment:
@fredriculrich, you'll be happy about this: by extracting the fetching logic out, we can now run the extension tests in a completely isolated domain context that will never request extensions from a server.